docs/writing-docs/doc-blocks.md

```diff
@@ -0,0 +1,370 @@
+
+<div style="background-color:#F8FAFC">
+TODO: ask tom if links are accurate
+</div>
+
+Props tables are rendered from an internal data structure called [ArgTypes](../api/stories#argtypes). When you declare a story's component metadata, Docs automatically extracts ArgTypes based on the component's properties.
+
+You can customize what's shown in the props table by customizing the ArgTypes data. This is currently available for [DocsPage](locate-docs-page) and `<Props story="xxx">` construct, but not for the `<Props of={component} />` construct.
+
+> NOTE: This API is experimental and may change outside of the typical semver release cycle
+
+The API documentation of ArgTypes is detailed in a [separate section](../api/stories#argtypes), but to control the description and default values, use the following fields:
+
+| Field                        | Description                                                                                    |
+|:-----------------------------|:----------------------------------------------------------------------------------------------:|
+| **name**                     |The name of the property                                                                        |
+| **type.required**            |The stories to be show, ordered by supplied name                                                |
+| **description**              |A markdown description for the property                                                         |
+|**table.type.summary**        |A short version of the type                                                                     |
+|**table.type.detail**         |A short version of the type                                                                     |
+|**table.defaultValue.summary**|A short version of the type                                                                     |
+|**table.defaultValue.detail** |A short version of the type                                                                     |
+|**control**                   |See [addon-controls README ](https://github.com/storybookjs/storybook/tree/next/addons/controls)|
+
+For instance:
+
+```js
+
+export default {
+  title: 'Button',
+  component: Button,
+  argTypes: {
+    label: {
+      description: 'overwritten description',
+      table: {
+        type: { 
+            summary: 'something short', 
+            detail: 'something really really long' 
+        },
+      },
+      control: {
+        type: null,
+      },
+    },
+  },
+};
+```
+
+This would render a row with a modified description, a type display with a dropdown that shows the detail, and no control.
+
+<div style="background-color:#F8FAFC">
+TODO: ask tom about this and how we can improve.
+
+My read on this is that we come from null that shows no control. But we have some shorthands that help the reader to show something.
+
+Probably we could add this as a aside, or give it more emphasis
```

Comment:
What are you thinking we should write @jonniebigodes?

Reply:
with the latest commit i reworded it. Let me know what you think. I went for clear and concise wording.

Reply:
LGTM

Reply:
Thanks for the feedback. Marking it as solved